src/commands/nickname.ts

```diff
@@ -0,0 +1,279 @@
+        const row = new MessageActionRow()
+            .addComponents(
+                new MessageButton()
+                    .setCustomId("nicknameVoteYes")
+                    .setLabel("Guter")
+                    .setStyle("SUCCESS"),
+                new MessageButton()
+                    .setCustomId("nicknameVoteNo")
+                    .setLabel("Lass ma")
+                    .setStyle("DANGER")
+            );
+        await command.reply({
+            content: `Eh Brudis, soll ich für ${user} ${nickname} hinzufügen?`,
+            components: [row]
+        });
+        const message = await command.fetchReply();
+        ongoingSuggestions.set(message.id, new Suggestion(user.id, nickname));
+        getUserVoteMap(message.id).set(user.id, new UserVote(Vote.YES, isTrusted(command.guild?.members.cache.get(user.id)!)));
+        return;
+    }
+
+
+    async updateNickName(user: GuildMember, nickname: string | null) {
+        await user.setNickname(nickname);
+    }
+}
+
+
+export class NicknameButtonHandler implements UserInteraction {
+    readonly ids = ["nicknameVoteYes", "nicknameVoteNo"];
+    readonly name = "NicknameButtonhandler";
+    readonly threshold = 5;
+
+
+    async handleInteraction(interaction: MessageComponentInteraction, client: Client): Promise<void> {
+        let suggestion = ongoingSuggestions.get(interaction.message.id);
+
+        if (suggestion === undefined) {
+            return interaction.update({
+                content: "Ich find den Namensvorschlag nicht. Irgend ein Huso muss wohl den Bot neugestartet haben. Macht am besten ne Neue auf",
+                components: []
+            });
+        }
+        let userVoteMap = getUserVoteMap(interaction.message.id);
+
+        const isTrust = isTrusted(interaction.guild?.members.cache.get(interaction.user.id)!);
+        if (!isTrust) {
+            return interaction.reply({
+                content: "Neeeeee, du bist nicht cool genug. Ich hab dein Vote nicht gezählt",
+                ephemeral: true
+            });
+        }
+        if (interaction.customId === "nicknameVoteYes") {
+            userVoteMap.set(interaction.user.id, new UserVote(Vote.YES, isTrust));
+        }
+        else if (interaction.customId === "nicknameVoteNo") {
+            userVoteMap.set(interaction.user.id, new UserVote(Vote.NO, isTrust));
+        }
+        // evaluate the Uservotes
+        let votes = Array.from(userVoteMap.values());
+        if (votes.filter(vote => vote.vote === Vote.NO).reduce((sum, uservote) => sum + uservote.getWeight(), 0) >= this.threshold) {
+            return interaction.update({
+                content: `Der Vorschlag: \`${suggestion.nickname}\` für <@${suggestion.nicknameUserID}> war echt nicht so geil`,
+                components: []
+            });
+        }
+        if (votes.filter(vote => vote.vote === Vote.YES).reduce((sum, uservote) => sum + uservote.getWeight(), 0) >= this.threshold) {
```

Comment:
Würde ich in eine Funktion oder Variable auslagern, damit klar ersichtlich ist was hier passiert. `hasEnoughVotes` oder so

Reply:
done